amplify/backend/function/fetchMotionState/src/index.js

```diff
@@ -3,37 +3,91 @@ const {
   getLatestMotionState,
   updateStakerRewardsInDB,
   getMotionData,
+  didMotionPass,
+  updateMotionMessagesInDB,
 } = require('./utils');
-
 /**
  * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
  */
 exports.handler = async (event) => {
   const { colonyAddress, transactionHash } = event.arguments?.input || {};
   /* Get latest motion state from chain */
   const motionData = await getMotionData(transactionHash);
-  const motionState = await getLatestMotionState(colonyAddress, motionData);
+  const motionState =
+    motionData && (await getLatestMotionState(colonyAddress, motionData));
   /*
    * Check if we need to update staker rewards
    * This ensures the rewards are present in the event a motion fails before going to a vote,
    * so the side that has objected can reclaim their stake.
    */
-  if (transactionHash && motionState === MotionState.Failed) {
-    if (motionData) {
-      const {
-        motionStakes: {
-          percentage: { yay: yayPercent },
-        },
-      } = motionData;
+  if (transactionHash && motionData && motionState === MotionState.Failed) {
+    const {
+      motionStakes: {
+        percentage: { yay: yayPercent },
+      },
+    } = motionData;
+
+    // Motion did not go to a vote
+    if (yayPercent !== '100') {
+      await updateStakerRewardsInDB(colonyAddress, transactionHash, motionData);
+    }
+  }
+
+  const didPass = didMotionPass(motionData);
+
+  if (
+    transactionHash &&
+    motionData &&
+    motionState === MotionState.Reveal &&
+    !motionData.inRevealPhase
+  ) {
+    await updateMotionMessagesInDB(
+      transactionHash,
+      motionData,
+      ['MotionRevealPhase'],
+      'inRevealPhase',
+    );
+  }
+
+  if (
+    transactionHash &&
+    motionData &&
+    (motionState === MotionState.Finalizable ||
+      motionState === MotionState.Finalized)
+  ) {
+    if (didPass && !motionData.hasPassed) {
```

Comment:
Some comments here would be handy for future us so we can quickly remember what each `if` really means